source/Calamari.Aws/Kubernetes/Discovery/AwsAuthenticationDetails.cs

```diff
@@ -33,28 +33,14 @@ public bool TryGetCredentials(ILog log, out AWSCredentials credentials)
             }
             else
             {
-                try
-                {
-                    // If not currently running on an EC2 instance,
-                    // this will throw an exception.
-                    credentials = new InstanceProfileAWSCredentials();
-                }
-                // Catching a generic Exception because AWS SDK throws undocumented exceptions.
-                catch (Exception instanceProfileException)
+                // The sequence of fallbacks trying to log in with credentials exposed by the worker.
+                // This follows the precedence document at https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-precedence
+                if (!(TryGetEnvironmentVariablesAwsCredentials(log, out credentials)
+                      || TryGetAssumeRoleWithWebIdentityCredentials(log, out credentials)
+                      || TryGetInstanceProfileAwsCredentials(log, out credentials)))
```

Comment:
Just a note that in my (untested) fix [I opted to use](https://github.com/OctopusDeploy/Calamari/pull/927/files#r1030943727) the `FallbackCredentialsFactory.GetCredentials()` in the AWS SDK which would give discovery all (?) AWS fallback options to the user in the default order as specified by AWS. It might be worth trying to see if it handles the case you're facing as it might save us from running into another edge case in the future.

Reply:
Nevermind, I just saw this is merged :) we can potentially use the above if another similar issue crops up again, but for now I think it's fine 👍